x-pack/plugins/uptime/public/components/monitor_management/monitor_config/monitor_config.tsx

```diff
@@ -37,46 +38,61 @@ export const MonitorConfig = () => {
   });
 
   const [testRun, setTestRun] = useState<TestRun>();
+  const [isTestRunInProgress, setIsTestRunInProgress] = useState<boolean>(false);
+  const [isFlyoutOpen, setIsFlyoutOpen] = useState<boolean>(false);
 
-  const onTestNow = () => {
+  const handleTestNow = () => {
     if (config) {
       setTestRun({ id: uuidv4(), monitor: config as MonitorFieldsType });
+      setIsTestRunInProgress(true);
+      setIsFlyoutOpen(true);
     }
   };
 
+  const handleTestDone = () => {
+    setIsTestRunInProgress(false);
+  };
```

Comment:
might be worth wrapping it in `useCallback` since it's being used as dependency in useEffect deep down.